### PR TITLE
Module 2

### PR DESCRIPTION
**Pull request by TheLegendaryPan**
Please check

### CODE REVIEW

File: app.py [hunk 1]
```diff
@@ -59,6 +65,93 @@ def updated():
...
+#        for result in resp['all']:
+#            myToDo.append({
+#                'CardID': result["idList"],
+#                'CardName': result["name"]
+#            })
+#        return render_template("trello-api.html", myToDo=myToDo)
+#    else: 
+#        return f"There is a problem with the request" 
+
+#@app.route('/ToDo/CreateCard')
```
Comment (Jonesey13) [inline app.py:146]: Removing the ability to add todos is a regression from Module 1 and will need to be working again for this to pass review.

File: app.py [hunk 2]
```diff
@@ -59,6 +65,93 @@ def updated():
...
+    return "Card moved to To Do, Please check Trello!"
+
+# return the list ID from resp_json_list for name "Done"
+# call update function from trellop to update the listID for cardID
+
+
+
+# list ID, card ID, card Name
+
+#getAll()
```
Comment (Jonesey13) [inline app.py:109]: As everything is in source control we shouldn't be leaving any code commented out here.

File: app.py [hunk 3]
```diff
@@ -1,13 +1,19 @@
 from flask import Flask, render_template, request, redirect, url_for
 import session_items as session
+import requests
```
Comment (Jonesey13) [inline app.py:3]: This dependency needs to be added to `requirements.txt` (are you running `setup.sh`? from the README instructions?)

File: templates/all_items.html [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+    <h1 class="display-4">To-Do App</h1>
+    <p class="lead">Just another to-do app.</p>
+  </div>
+  <div class="row">
+    <div class="col-md-12">
+      <h2>Current To Do Items from Trello:</h2>
+           {% for cards_from_todo in cards_from_todo %}
+           <form action="{{url_for('complete_item')}}" method="POST">
+            <input type ='text' name='card_name' value="{{cards_from_todo['name']}}" readonly="readonly">
+            <input type = 'text' name="list_id" value="{{cards_from_todo['idList']}}" readonly="readonly">
```
Comment (Jonesey13) [inline templates/all_items.html:15]: These 2 input fields do not need to be displayed to the end user.

File: app.py [hunk 5]
```diff
@@ -59,6 +65,93 @@ def updated():
...
+
+
+@app.route('/complete_item', methods = ['POST', 'GET'])
+def complete_item():
+    if request.method == 'POST':
+        card_name = request.form['card_name']
+        card_id = request.form['card_id']
+        
+        trelloapp.move_card_to_done(card_id)
+    return "Card moved to Done, Please check Trello!"
```
Comment (Jonesey13) [inline app.py:91]: The end user should not be aware that the backend is calling Trello.

File: trelloapp.py [hunk 6]
```diff
@@ -0,0 +1,124 @@
+import requests
+import os
+from flask import Flask, render_template
+import json
+from dotenv import load_dotenv  #to invoke .env file
```
Comment (Jonesey13) [inline trelloapp.py:5]: If you're running flask via `flask run` you do not need to run `load_dotenv()` manually.

File: app.py [hunk 7]
```diff
@@ -1,13 +1,19 @@
 from flask import Flask, render_template, request, redirect, url_for
 import session_items as session
+import requests
+import os
+import trelloapp
+import json
 
 app = Flask(__name__)
 app.config.from_object('flask_config.Config')
```
Comment (Jonesey13) [inline app.py:9]: This line (and the `flask_config.py` file) is no longer needed (as we should longer be using the flask session state)

File: app.py [hunk 8]
```diff
@@ -1,13 +1,19 @@
...
 
 @app.route('/') #redirect root to the url or route mapped to the index function
 def root():
-    return redirect(url_for('index'))
+    return redirect(url_for('getAll')) ## replaced index
 
+## All route from below up to module is NO LONGER applicable since replacing index 
+## with getAll from Module 2. Keep for reference No longer calling session.py
 @app.route('/items', methods = ["GET", "POST", "PUT"])
```
Comment (Jonesey13) [inline app.py:17]: This route (and other calling `session`) should be removed (as we're using trello to back the todos and not the flask session cookie)

File: trelloapp.py [hunk 9]
```diff
@@ -0,0 +1,124 @@
...
+load_dotenv()
+
+trelloapp = Flask(__name__)
+
+#Module2 project #
+#APP_KEY = os.environ.get('APP_KEY')  # take env variable from local system envor
+#APP_TOKEN=os.environ.get('APP_TOKEN')
+
+APP_KEY = os.getenv("APP_KEY")  # take .env from dotenv
```
Comment (Jonesey13) [inline trelloapp.py:14]: How these are wired up should be documented in the README

File: trelloapp.py [hunk 10]
```diff
@@ -0,0 +1,124 @@
...
+APP_TOKEN = os.getenv("APP_TOKEN")
+
+board_id= '5ef1186ddea5ff1b03e085e6'
```
Comment (Jonesey13) [inline trelloapp.py:17]: How to setup the board id should be documented in the README

File: trelloapp.py [hunk 11]
```diff
@@ -0,0 +1,124 @@
...
+list_id_ToDo = '5ef1186d26a8d939ea575069'
```
Comment (Jonesey13) [inline trelloapp.py:18]: These 3 list ids could be inferred by name (with the special names documented in the README)

File: trelloapp.py [hunk 12]
```diff
@@ -0,0 +1,124 @@
...
+list_id_Pending = '5ef1186d2fd57d026f03add0'
+list_id_Done = '5ef1186dcbda554f16c6d66f'
+
+def get_card():
```
Comment (Jonesey13) [inline trelloapp.py:22]: Guessing this method is unused as it relies on a magic card id

File: app.py (hunk 7 above)
Comment (Jonesey13) [reply app.py:9]: Same for `session_items.py` and the SECRET_KEY env variable (both can go)

**Review by Jonesey13 — COMMENTED** [review]
The integration with Trello appears to be working for changing statuses but there's no functionality to add cards which a requirement of this exercise.

There's also a missing dependency `requests` in the setup.

Also the old code that uses the flask session state should be purged (there's alot of unused code floating around which will make this hard to maintain going forward)

File: app.py (hunk 2 above)
Comment (TheLegendaryPan) [reply app.py:109]: Removed commented out codes. thanks!

File: app.py (hunk 3 above)
Comment (TheLegendaryPan) [reply app.py:3]: Got it. I've updated the requirements.txt with requests==2.23.0
Good point.

File: templates/all_items.html (hunk 4 above)
Comment (TheLegendaryPan) [reply templates/all_items.html:15]: Will remove once I fix the add card feature back. Was using this to see that I was pulling the correct card & list details.

File: app.py (hunk 5 above)
Comment (TheLegendaryPan) [reply app.py:91]: Done. Thanks!

File: trelloapp.py (hunk 9 above)
Comment (TheLegendaryPan) [reply trelloapp.py:14]: Documentation included in README. Thanks!

File: trelloapp.py (hunk 10 above)
Comment (TheLegendaryPan) [reply trelloapp.py:17]: Instructions for pulling board and list IDs are included in README. Thanks!

File: trelloapp.py (hunk 12 above)
Comment (TheLegendaryPan) [reply trelloapp.py:22]: that's correct. was using that as a proof of concept to test getting card details.

**Pull request merged**